Managers/MenuMan.cpp

```diff
@@ -0,0 +1,264 @@
+#include "MenuMan.h"
+#include "SettingsMan.h"
+#include "FrameMan.h"
+#include "UInputMan.h"
+#include "PresetMan.h"
+#include "ConsoleMan.h"
+#include "MetaMan.h"
+
+#include "GUI.h"
+#include "AllegroScreen.h"
+#include "AllegroBitmap.h"
+#include "AllegroInput.h"
+
+#include "Controller.h"
+#include "TitleScreen.h"
+#include "MainMenuGUI.h"
+#include "ScenarioGUI.h"
+#include "MetagameGUI.h"
+#include "LoadingScreen.h"
+
+#include "NetworkServer.h"
+
+namespace RTE {
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void MenuMan::Initialize(bool firstTimeInit) {
+		m_ActiveMenu = ActiveMenu::MenusDisabled;
+
+		m_GUIScreen = std::make_unique<AllegroScreen>(g_FrameMan.GetBackBuffer32());
+		m_GUIInput = std::make_unique<AllegroInput>(-1, g_UInputMan.GetJoystickCount() > 0);
+
+		if (firstTimeInit) {
+			g_LoadingScreen.Create(m_GUIScreen.get(), m_GUIInput.get());
+			g_PresetMan.LoadAllDataModules();
+			g_LoadingScreen.Destroy();
+
+			// Load the different input device icons. This can't be done during UInputMan::Create() because the icon presets don't exist so we need to do this after modules are loaded.
+			g_UInputMan.LoadDeviceIcons();
+
+			if (g_NetworkServer.IsServerModeEnabled()) { g_SettingsMan.SetSkipIntro(true); }
+		}
+		m_TitleScreen = std::make_unique<TitleScreen>(m_GUIScreen.get());
+		m_MainMenu = std::make_unique<MainMenuGUI>(m_GUIScreen.get(), m_GUIInput.get());
+		m_ScenarioMenu = std::make_unique<ScenarioGUI>(m_GUIScreen.get(), m_GUIInput.get());
+
+		// TODO: MetaGameGUI doesn't seem to actually do anything with the Controller but removing conflicts with the second Create() method so that needs to be sorted out sometime in the year 3000.
+		m_MenuController = std::make_unique<Controller>(Controller::CIM_PLAYER);
+		g_MetaMan.GetGUI()->Create(m_MenuController.get());
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void MenuMan::Reinitialize() {
+		g_ConsoleMan.Destroy();
```

Comment:
I have to disagree on everything.
MenuMan is very much responsible for all this stuff, because all this stuff *is* menu stuff.
You kinda forgot that the biggest pain in the ass of resolution changing is GUIs.
This can be solved but it requires a bunch of work inside the GUILib.

Honestly tho, I could probably eliminate `DestroyTempBackBuffers` by doing some stuff that may or may not be questionable hacky shit.

Also you're forgetting that ConsoleMan is also a GUI. The fact it's a singleton and is being updated separately in and out of game doesn't change that fact.
So I have to deal with it here, otherwise resolution changing goes all over the place again.

MetaMan is a piece of shit, yes.
But even if it wasn't this would probably be the same because it owns it's GUI and it needs to be set somewhere.

Reply:
My understanding of how this stuff all works is far from perfect, so yeah, I may very well be wrong in what I'm saying.

Just from a code-design perspective, it seems to me that destroying frame man's temp back buffers isn't the job of a menu manager, cause menu man shouldn't really have to know details about frame man and how that shit works. It should ideally just be told what it needs to do its job, and tell whoever needs to be told about resolution changes that the resolution has changed, and let that do its job.
But ideal != reality, sometimes you have coupling in code, and if it's difficult to change program flow to do this, that's not the end of the world.

As for consoleman, fair point, you're right that I wasn't thinking about this being for resolution changes, and I guess this is the best place to fix console man's size after a resolution change, since nothing else really knows about it? Maybe frameman could do it, but maybe that'd make less sense.
And yeah, metaman is in the same boat here.